v2/yarpcgrpc/outbound.go

```diff
@@ -144,34 +112,33 @@ func (o *Outbound) invoke(
 	if responseMD != nil {
 		callOptions = []grpc.CallOption{grpc.Trailer(responseMD)}
 	}
-	apiPeer, onFinish, err := o.peerChooser.Choose(ctx, request)
+	apiPeer, onFinish, err := o.Chooser.Choose(ctx, request)
 	if err != nil {
 		return err
 	}
 	defer func() { onFinish(retErr) }()
 	grpcPeer, ok := apiPeer.(*grpcPeer)
 	if !ok {
-		return peer.ErrInvalidPeerConversion{
+		return yarpcpeer.ErrInvalidPeerConversion{
 			Peer:         apiPeer,
 			ExpectedType: "*grpcPeer",
 		}
 	}
 
-	tracer := o.t.options.tracer
-	createOpenTracingSpan := &transport.CreateOpenTracingSpan{
-		Tracer:        tracer,
+	createOpenTracingSpan := &yarpctracing.CreateOpenTracingSpan{
```

Comment:
optional: pulling the tracing logic out to a separate function might help clean this long code up a bit